api/v1alpha1/imageclusterinstall_webhook.go

```diff
@@ -81,20 +69,43 @@ func (r *ImageClusterInstall) ValidateUpdate(old runtime.Object) (admission.Warn
 		return nil, fmt.Errorf("old object is not an ImageClusterInstall")
 	}
 
-	if oldClusterInstall.Spec.BareMetalHostRef == nil && r.Spec.BareMetalHostRef == nil {
-		return nil, nil
-	}
-
 	// Allow update if it's not the spec
 	if !isSpecUpdate(oldClusterInstall, r) {
 		return nil, nil
 	}
-	if BMHRefsMatch(oldClusterInstall.Spec.BareMetalHostRef, r.Spec.BareMetalHostRef) {
-		return nil, fmt.Errorf("Cannot update ImageClusterInstall when BareMetalHostRef is set, unset BareMetalHostRef before making changes")
+	// block update if the installation started
+	if installationStarted(oldClusterInstall) {
+		return nil, fmt.Errorf("cannot update ImageClusterInstall when the configImage is ready")
 	}
 	return nil, nil
 }
 
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *ImageClusterInstall) ValidateDelete() (admission.Warnings, error) {
+	return nil, nil
+}
+
+func (r *ImageClusterInstall) validate() error {
+	if err := isValidSSHPublicKey(r.Spec.SSHKey); err != nil {
+		return fmt.Errorf("invalid ssh key: %v", err)
+	}
+	if err := isValidHostname(r.Spec.Hostname); err != nil {
+		return fmt.Errorf("invalid hostname: %w", err)
+	}
+
+	if err := isValidMachineNetwork(r.Spec.MachineNetwork); err != nil {
+		return fmt.Errorf("invalid machine network: %w", err)
+	}
+	return nil
+}
+
+func installationStarted(r *ImageClusterInstall) bool {
+	// Block update if the image is ready
+	// Consider updating this to check the ClusterInstallCompleted condition
+	imageReadyCondition := findCondition(r.Status.Conditions, hivev1.ClusterInstallRequirementsMet)
```

Comment:
hmm, i am not sure we update this condition anymore to say the truth, I think better to validate if there is bmh in status or even in reference

Reply:
Or we can add this condition here https://github.com/openshift/image-based-install-operator/blob/main/controllers/imageclusterinstall_controller.go#L277